api/cryptocom/requests.go

```diff
@@ -0,0 +1,168 @@
+package cryptocom
+
+import (
+	"database/sql"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/shopspring/decimal"
+)
+
+func (c *Client) AuthRequest() *Request {
+	r := &Request{
+		Id:     1,
+		Type:   AuthRequest,
+		Method: "public/auth",
+		ApiKey: c.key,
+		Nonce:  generateNonce(),
+	}
+
+	c.generateSignature(r)
+	return r
+}
+
+func (c *Client) subscribeRequest(channels []string) *Request {
+	return &Request{
+		Id:     1,
+		Type:   SubscribeRequest,
+		Method: "subscribe",
+		Params: map[string]interface{}{"channels": channels},
+		Nonce:  generateNonce(),
+	}
+}
+
+func (c *Client) hearBeatRequest(reqId int) *Request {
+	return &Request{
+		Id:     reqId,
+		Type:   HeartBeat,
+		Method: "public/respond-heartbeat",
+	}
+}
+
+func (c *Client) createOrderRequest(
+	reqID int,
+	ask string,
+	bid string,
+	orderSide string,
+	orderType string,
+	price decimal.Decimal,
+	volume decimal.Decimal,
+	uuid uuid.UUID) *Request {
+	return &Request{
+		Id:     reqID,
+		Type:   OrderRequest,
+		Method: "private/create-order",
+		Params: map[string]interface{}{
+			"instrument_name": strings.ToUpper(ask) + "_" + strings.ToUpper(bid),
+			"side":            strings.ToUpper(orderSide),
+			"type":            strings.ToUpper(orderType),
+			"price":           price,
+			"quantity":        volume,
```

Comment:
https://exchange-docs.crypto.com/#private-create-order 
Check quantity and notional parameters.
In case of Market BUY order, we have to pass notional instead of amount.